### PR TITLE
fix: small typos in documentation

### PR DESCRIPTION
**Pull request by sanjom**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/frontend-system/architecture/03-extensions.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ Note that while the `createExtension` is public API and used in many places, it
...
+Communication between extensions happens in one direction, from one child extension through the attachment point to its parent. The child extension outputs data which is then passes as inputs to the parent extension. This data is called Extension Data, where the shape of each individual piece of data is described by an Extension Data Reference. These references are created separately from the extensions themselves, and can be shared across multiple different kinds of extensions. Each reference consists of an ID and a TypeScript type that the data needs to conform to, and represents one type of data that can be shared between extensions.
```
Comment (benjdlambert) [inline docs/frontend-system/architecture/03-extensions.md:86]: I think this one was correct :pray:

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Thanks for the cleanup! There's just one thing that doesn't make sense I think! 👍

File: docs/frontend-system/architecture/03-extensions.md (hunk 1 above)
Comment (af-jml) [reply docs/frontend-system/architecture/03-extensions.md:86]: You're right, I misread it as "which it then passes". Will fix it.

**Review by vinzscam — APPROVED** [review]

**Comment by freben** [comment]
Hey, remember that you have to DCO sign all commits for us to have the ability to merge them. Check out the bot comment above for instructions!

**Comment by af-jml** [comment]
Thanks @freben, could you re-approve the PR and merge it? I updated the commits.

**Comment by sanjom** [comment]
Thanks @freben, could you re-approve the PR and merge it? I updated the commits.

**Review by freben — APPROVED** [review]

**Pull request merged**